core/src/components/datetime/test/disable-dates/e2e.ts

```diff
@@ -0,0 +1,166 @@
+import type { E2EPage } from '@stencil/core/testing';
+import { newE2EPage } from '@stencil/core/testing';
+
+describe('datetime: disable dates', () => {
+
+  describe('return values', () => {
+
+    let page: E2EPage;
+
+    beforeEach(async () => {
+      page = await newE2EPage({
+        html: '<ion-datetime value="2021-10-01"></ion-datetime>'
+      });
+    });
+
+
+    describe('when isDateEnabled returns true', () => {
+
+      it('calendar days should be enabled', async () => {
+        await page.$eval('ion-datetime', (el: any) => {
+          el.isDateEnabled = () => true;
+        });
+
+        await page.waitForChanges();
+
+        const disabledDays = await page.findAll('ion-datetime >>> .calendar-day[disabled]:not(.calendar-day-padding)');
```

Comment:
Could we break this logic out into a util function that returns an array of disabled days (and then check the length when just the number is needed)? That sort of pattern is used a bunch here.